src/app/shared/components/wallet-modal/wallet-modal.component.ts

```diff
@@ -0,0 +1,19 @@
+import { Component, ViewChild } from "@angular/core";
+import { ModalComponent } from "@shared/components/modal/modal.component";
+
+@Component({
+    selector: "app-wallet-modal",
+    templateUrl: "./wallet-modal.component.html",
+    styleUrls: ["./wallet-modal.component.scss"]
+})
+export class WalletModalComponent {
+    @ViewChild("modal") modal: ModalComponent | undefined = undefined;
```

Comment:
if a union type for undefined is passed there is no need to define an initial value